scripts/release_scripts/run_macaron.sh

```diff
@@ -140,6 +147,64 @@ function assert_path_exists() {
     fi
 }
 
+# Create a directory if it does not exist.
+# Arguments:
+#   $1: The directory to create.
+function create_dir_if_not_exists() {
+    dir=$1
+    if [ ! -d "$dir" ]; then
+        mkdir --parents "$dir"
+    fi
+}
+
+# Add a directory to the list of volume mounts stored in the ``mounts`` global variable.
+#
+# Arguments:
+#   $1: The macaron argument from which the directory is passed into this script.
+#   $2: The path to the directory on the host.
+#   $3: The path to the directory inside the container.
+#   $4: Mount option. Note: this MUST be either `ro,Z` for readonly volume mounts,
+#       or `rw,Z` otherwise.
+function mount_dir() {
```

Comment:
It would be good to pass an option to create the directory, which is disabled by default. This would make us explicitly enable directory creation at callsites. Otherwise, it is easy to call this function thinking we are just mounting a directory, but end up creating it on the host by mistake.

Reply:
In b8babbae18e20bedc7568d62a894227a2d2fcc0e, I create 3 functions for mounting dirs:
* `mount_dir_ro`
* `mount_dir_rw_allow_create`
* `mount_dir_rw_forbid_create`
I think the last 2 are more readable than having an additional param and calling the `mount_dir` function like this:
```
mount_dir_rw [...] 0
mount_dir_rw [...] 1
```
where `0` and `1` are values for whether directory creation is allowed or not.